### PR TITLE
Update CI to not use http's private CI images

### PR DESCRIPTION
**Pull request by SeanTAllen**
We don't reach into any other repo other than shared-docker to
get CI images for use. See https://github.com/ponylang/contributors/blob/master/ci/ci-image-organization.md
for more information.

### CODE REVIEW

File: .ci-dockerfiles/x86-64-unknown-linux-builder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ARG FROM_TAG=release
+FROM ponylang/shared-docker-ci-x86-64-unknown-linux-builder:${FROM_TAG}
+
+RUN apk update \
+  && apk upgrade \
+  && apk upgrade curl \
+  && apk add --update libressl-dev pcre2-dev
```
Comment (mfelsche) [inline .ci-dockerfiles/x86-64-unknown-linux-builder/Dockerfile:7]: pcre2-dev is not necessary anymore as we don't depend on regex anymore.

**Comment by mfelsche** [comment]
Instead of creating a new image, could we also use https://hub.docker.com/repository/docker/ponylang/shared-docker-ci-x86-64-unknown-linux-builder-with-ssl ?

I had a changeset locally to use this, but it got lost during rebase i think.

**Comment by SeanTAllen** [comment]
@mfelsche if we only need ssl now then yes, I will update the PR.

**Comment by SeanTAllen** [comment]
@mfelsche PR updated.

File: .github/workflows/ci-image-updates.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: CI image updates
```
Comment (mfelsche) [inline .github/workflows/ci-image-updates.yml:1]: we don't need this anymore, right?

Comment (SeanTAllen) [reply .github/workflows/ci-image-updates.yml:1]: ooops missed that

**Comment by SeanTAllen** [comment]
@mfelsche updated again

**Comment by mfelsche** [comment]
Thx! :)

**Pull request merged**